### DISCUSSION TITLE
关于当前dev分支签名服务器问题可以到这里提问

### DISCUSSION THREAD

**Discussion by KomeiDiSanXian** [Mrs4s/go-cqhttp#2314] [General]
### ~~叠甲声明~~
---
我会尽力帮大家查看问题在哪，受限于本人的技术问题，一些问题我可能无法解决，这就需要大家的力量帮忙解决，感谢各位的大力支持了
### 一些解释
---
以下内容提及的`dev` 均指 526391e

### 常见问题: 
---
1.  类似 #2310 当前的[qsign](https://github.com/fuqiuluo/unidbg-fetch-qsign) (截至到v1.1.6) 基于Android平台计算参数，所以不支持其他协议(iPad)。推荐你更改`device.json` 中的`protocol` 字段为 `6` 以使用qsign

2.  当前的dev 需要**在qsign启动之后才能启动gocq**，否则会注册实例失败

3.  类似 #2303 如果qsign崩溃了，当前dev的gocq也同样需要重启来进行重新注册实例，即使你**开启**了qsign的自动注册功能也**需要重新启动**

4. 当前dev 版本对`config.yml` 做出了更改，如果你是从先前的版本直接覆盖更新，请注意更新`config.yml`。你也可以选择删除当前的`config.yml` 重新生成一份并进行修改

5.  qsign版本低于v1.1.0也可使用当前dev 版本，只需要在`config.yml` 内，在 `sign-server` (就是你配置签名服务器的那个) 字段下修改`is-below-110` 字段并设置为`true` 就可以了。具体的前往[这里](https://github.com/Mrs4s/go-cqhttp/blob/dev/modules/config/default_config.yml)查看并仿照更改

6.  类似 fuqiuluo/unidbg-fetch-qsign#144 和 fuqiuluo/unidbg-fetch-qsign#134 的问题，这是因为gocq设置的请求超时时间为5秒，超过这一时间就会报错`context deadline exceeded`，但qsign那边收到请求仍会进行处理，只是gocq这边会接收不到返回的消息

### 目前找到的和想到的问题就是这些，欢迎大家在下面提问

**Comment by scriptxiaozi** [comment]
是不是目前只支持到8.9.63版本

**Reply by KomeiDiSanXian** [reply]
目前，gocq对于协议的支持似乎是写死在代码里的，在
MiraiGo/client/internal/auth/auth.go里可以看到
对于apad协议，如下
```go
AndroidPad: {
			ApkId:           "com.tencent.mobileqq",
			AppId:           537164888,
			SubAppId:        537164888,
			AppKey:          "0S200MNJT807V3GE",
			SortVersionName: "8.9.63.11390",
			BuildTime:       1685069178,
			ApkSign:         []byte{0xA6, 0xB7, 0x45, 0xBF, 0x24, 0xA2, 0xC2, 0x77, 0x52, 0x77, 0x16, 0xF6, 0xF3, 0x6E, 0xB6, 0x8D},
			SdkVersion:      "6.0.0.2546",
			SSOVersion:      20,
			MiscBitmap:      150470524,
			SubSigmap:       0x10400,
			MainSigMap: WLOGIN_A5 | WLOGIN_RESERVED | WLOGIN_STWEB | WLOGIN_A2 | WLOGIN_ST |
				WLOGIN_LSKEY | WLOGIN_SKEY | WLOGIN_SIG64 | 1<<16 | WLOGIN_VKEY | WLOGIN_D2 |
				WLOGIN_SID | WLOGIN_PSKEY | WLOGIN_AQSIG | WLOGIN_LHSIG | WLOGIN_PAYTOKEN, // 16724722
			QUA:      "V1_AND_SQ_8.9.63_4194_YYB_D",
			Protocol: AndroidPad,
		},
```
所以目前的gocq只支持到8.9.63

**Comment by zhaodice** [comment]
有最新的协议字符串嘛~我看qsign都支持到8.9.70了

**Reply by KomeiDiSanXian** [reply]
这个等gocq更新吧，使用了另外的仓库再等等吧

**Reply by zhaodice** [reply]
不知道为什么，即使是用了sign，个别一些人依然错误代码235，重试n次就升级为237了，即使是，我用了最新的8.9.70协议和sign，太奇怪了（反正我自己是登陆成功的了）

**Comment by SuperKenVery** [comment]
说起来 感觉dev分支挺稳定的（）为什么还没有合并进main呢

以及等miraigo更新支持新版qsign（）我还等着matrix-qq-bridge更新呢、、、、

**Comment by zhaodice** [comment]
> 说起来 感觉dev分支挺稳定的（）为什么还没有合并进main呢
> 
> 以及等miraigo更新支持新版qsign（）我还等着matrix-qq-bridge更新呢、、、、

自己编译也未尝不可

**Comment by Arymodu** [comment]
本地签名服务器部署成功后我使用协议1和6均可以正常登录，但是群聊一直处于被风控的状态（过了24h也没有解除），使用协议2群聊可以正常发送消息但是会和之前一样过一段时间就风控（私聊均正常）。翻遍了gocq和qsign的issue似乎都没有好的解决办法，删除文件重新登陆还是一样的结果，开摆。

**Reply by zhaodice** [reply]
gocq对接qsign没对接完全，没有实现ssoPackets（也就是说，tx需要发包给so文件，so文件再回复，gocq没有实现这个桥梁）

**Comment by rhwong** [comment]
问问最新dev里的sign-server-bearer项是什么？在qsign里没找到。

**Reply by 1umine** [reply]
一种访问授权方式，里面一般填服务器给你的访问token等，qsign没设置目前用不到，不用管，用别人的签名服务器可能有的人为了控制下服务器压力可能会需要授权，用本地搭的就不用管了

**Reply by rhwong** [reply]
如果我搭建公共的api，我该怎么设置sign-server-bearer呢？

**Reply by 1umine** [reply]
qsign 好像没有设置，只有 key 控制访问，要设置bearer token只能自己设置代理或者什么的 （

**Reply by rhwong** [reply]
原来是这样（）

**Reply by rhwong** [reply]
事实上我想基于IP限制同一个IP只允许一个uin使用我的公共qsign服务，因为我发现有一个ip登了4个号全都是同样的头像同样的昵称，我担心用于黑产。但是又想不到什么好办法做限制。

**Comment by zhaodice** [comment]
最新的gocq有问题。。。就是把协议改成8.9.70后自己编译，就没法正常登陆，说什么decrypt failed，我都不好意思issue毕竟是改了它源码自己编译的。。怀疑是sso包在升级版本后识别失败

**Comment by MuuuShin** [comment]
怎么解决第六条context deadline exceeded的问题呢？被卡在45出不去了

**Reply by KomeiDiSanXian** [reply]
当时的dev版本是不能改超时时间的，现在的可以改了，试着改把超时时间改大一点？
```yaml
  # 签名服务请求超时时间(s)
  sign-server-timeout: 60
```
现在的默认是60s的，看看是不是被改过了

**Reply by MuuuShin** [reply]
原来如此，我当初这条路走不通了后转向了另一篇 切了协议到70后就好了

**Comment by ability66** [comment]
请问我现在使用linux配置好了签名服务器，但是dev分支的gocq编译后检查不到这个端口该怎么解决呢
![image](https://github.com/Mrs4s/go-cqhttp/assets/106503132/e45b9aed-06d5-4281-af67-083aa8db115e)
显示已经在监听这个端口了
![image](https://github.com/Mrs4s/go-cqhttp/assets/106503132/6c382dcc-e6a7-4adc-bf88-b427c13216db)
签名服务器端口也是一致的，但是直接运行gocq就会这样
![Uploading image.png…]()

**Reply by ability66** [reply]
![image](https://github.com/Mrs4s/go-cqhttp/assets/106503132/0bf45d5a-3aaf-4a2c-9d4e-c851a495ed1f)
这是第三zhang张图

**Reply by 1umine** [reply]
配置改了 重新生成个配置

**Reply by ability66** [reply]
![image](https://github.com/Mrs4s/go-cqhttp/assets/106503132/27ea34fb-5566-4b76-a074-e0fbcb5e230a)
改了配置，但是还是45，是因为我账号风险等级太高了吗

**Reply by MuuuShin** [reply]
63协议已经没法正常使用了，你该做的是将协议切换为70以上的版本

**Comment by aknyzsd** [comment]
在没有使用dev分支是能用qsign服务器的，但是会报错t554，看了一下需要换成dev分支，换完之后出现以下报错

[2023-10-04 13:21:03] [INFO]: 当前版本:07214e3
[2023-10-04 13:21:03] [INFO]: 将使用 device.json 内的设备信息运行Bot.
[2023-10-04 13:21:04] [INFO]: 检查签名服务器：http:127.0.0.1:19191  (1/2)
[2023-10-04 13:21:04] [INFO]: 检查签名服务器：-  (2/2)
[2023-10-04 13:21:04] [WARNING]: 签名服务器 http:127.0.0.1:19191 可能不可用，请求出现错误：Get "http:127.0.0.1:19191": http: no Host in request URL
[2023-10-04 13:21:04] [WARNING]: no usable sign server
[2023-10-04 13:21:04] [WARNING]: 警告: 未配置签名服务器或签名服务器不可用, 这可能会导致登录 45 错误码或发送消息被风控
[2023-10-04 13:21:04] [INFO]: Bot将在5秒后登录并开始信息处理, 按 Ctrl+C 取消.
[2023-10-04 13:21:09] [INFO]: 开始尝试登录并同步消息...
[2023-10-04 13:21:09] [INFO]: 使用协议: Android Pad 8.9.63.11390
[2023-10-04 13:21:09] [INFO]: 正在检查协议更新...
[2023-10-04 13:21:16] [INFO]: Protocol -> connect to server: 42.81.179.212:8080
[2023-10-04 13:21:17] [WARNING]: 登录需要滑条验证码, 请验证后重试.
[2023-10-04 13:21:17] [WARNING]: 请选择提交滑块ticket方式:
[2023-10-04 13:21:17] [WARNING]: 1. 自动提交
[2023-10-04 13:21:17] [WARNING]: 2. 手动抓取提交
[2023-10-04 13:21:17] [WARNING]: 请输入(1 - 2)：

**Reply by 1umine** [reply]
建议仔细看看报错说了什么

**Reply by aknyzsd** [reply]
> 建议仔细看看报错说了什么
起猛了，没加斜杠
